hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/util/StreamerUtil.java

```diff
@@ -163,6 +166,27 @@ public static HoodieIndexConfig getIndexConfig(Configuration conf) {
         .build();
   }
 
+  /**
+   * Returns the timeGenerator config with given configuration.
+   */
+  public static HoodieTimeGeneratorConfig getTimeGeneratorConfig(Configuration conf) {
+    HoodieTimeGeneratorConfig.Builder builder = HoodieTimeGeneratorConfig.newBuilder()
+        .withPath(conf.getString(FlinkOptions.PATH))
+        .fromProperties(flinkConf2TypedProperties(conf));
+
+    if (OptionsResolver.isLockRequired(conf) && !conf.containsKey(HoodieLockConfig.LOCK_PROVIDER_CLASS_NAME.key())) {
+      // configure the fs lock provider by default
```

Comment:
Do we need this? Shouldn't the lock provider already be configured in the configuration? Or maybe
`HoodieWriteClients.getHoodieClientConfig(Configuration).getHoodieTimeGeneratorConfig()` should work?